DFE-Digital/early-careers-framework
# NPQ profiles API endpoint refactor

### Context

- There are changes that need to starting happening to the `api/v1/npq-profiles` endpoint
- The endpoint implemented `JSONAPI::ActsAsResourceController` from a third party gem
- This gem although brought a lot to the table out of the box is extraordinarily constraining and difficult to work with when try to extend it
- This will now make the endpoint more easily changeable for the next set of changes that need to happen

### Changes proposed in this pull request

-  `api/v1/npq-profiles` now uses (mostly) vanilla rails
- Removed the gem `jsonapi-resources` as it is no longer used

### Guidance to review

- There should be little to no behavioural change

### Testing

- Can test in review app
- Or you can fire up `npq-registration` locally and should be able to register and see new data created in ECF

### Review Checks

### How can I view this in a review app?

- Create or find an existing `NPQRegistrationApiToken`
- Create or find an existing user
- Call the endpoint
```
curl -X POST -H "Authorization: Bearer TOKEN" -H "Content-Type: application/vnd.api+json" -d '{"data":{"type":"npq_profiles","attributes":{"teacher_reference_number":"1234567","teacher_reference_number_verified":true,"date_of_birth":"1990-12-13","school_urn":"123456","headteacher_status":"no"},"relationships":{"user":{"data":{"type":"users","id":"USER_ID_HERE"}},"npq_lead_provider":{"data":{"type":"npq_lead_providers","id":"NPQ_LEAD_PROVIDER_ID"}},"npq_course":{"data":{"type":"npq_courses","id":"NPQ_COURSE_ID"}}}}}' https://ecf-review-pr-847.london.cloudapps.digital/api/v1/npq-profiles
```
- Should see new `NPQValidationData` record persisted
